src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,755 @@
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.CLOCK_STC_AVG.getName(), checkForNullData(decoderStats.getClockStcAvg()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.HDR_TYPE_IN.getName(), checkForNullData(decoderStats.getHdrTypeIn()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.HDR_TYPE.getName(), checkForNullData(decoderStats.getHdrType()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.HDR_COLOUR_PRIMARIES.getName(), checkForNullData(decoderStats.getHdrColourPrimaries()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.HDR_TRANSFER_CHARACTERISTICS.getName(), checkForNullData(decoderStats.getHdrTransferCharacteristics()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.HDR_MATRIX_COEFFICIENTS.getName(), checkForNullData(decoderStats.getHdrMatrixCoefficients()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.TC_RECEIVED_PACKETS.getName(), checkForNullData(decoderStats.getTcReceivedPackets()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.TC_OUTPUT_PACKETS.getName(), checkForNullData(decoderStats.getTcOutputPackets()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.TC_FREED_PACKETS.getName(), checkForNullData(decoderStats.getTcFreedPackets()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.BUFFERING_DELAY.getName(), checkForNullData(decoderInfo.getBufferingDelay()));
+					stats.put(decoderStatisticGroup + DecoderMonitoringMetric.LATENCY.getName(), checkForNullData(decoderInfo.getLatency()));
+
+				} else {
+					updateDecoderStatisticsFailedMonitor(failedMonitor, decoderID);
+				}
+			} else {
+				updateDecoderStatisticsFailedMonitor(failedMonitor, decoderID);
+			}
+		} catch (Exception e) {
+			updateDecoderStatisticsFailedMonitor(failedMonitor, decoderID);
+		}
+	}
+
+	/**
+	 * Update failedMonitor with Getting stream statistics error message
+	 *
+	 * @param failedMonitor list statistics property
+	 */
+	private void updateStreamStatisticsFailedMonitor(Map<String, String> failedMonitor) {
+		failedMonitor.put(MonitoringMetricGroup.STREAM_STATISTICS.getName(), DecoderConstant.GETTING_STREAM_STATS_ERR);
+	}
+
+	/**
+	 * This method is used update stream statistic from DTO
+	 *
+	 * @param stats list statistics property
+	 * @param streamInfo List stream config properties
+	 * @param streamStats List stream statistic properties
+	 * @param srt List SRT
+	 */
+	private void updateStreamStats(Map<String, String> stats, StreamInfo streamInfo, StreamStats streamStats, SRT srt) {
+		String streamStatisticGroup = MonitoringMetricGroup.STREAM_STATISTICS.getName() + DecoderConstant.SPACE + streamInfo.getName() + DecoderConstant.HASH;
+
+		// Update static stats
+		stats.put(streamStatisticGroup + StreamMonitoringMetric.ID.getName(), checkForNullData(streamInfo.getId()));
```

Comment:
The same story for the issue above, please add a for loop and move this logic to the new method `streamInfo.getValueByStreamMonitoringMetric()`